lib/internal/modules/esm/module_job.js

```diff
@@ -92,6 +93,21 @@ class ModuleJob {
         this.module.instantiate();
       }
     } catch (e) {
+      if (e.message.includes('does not provide an export named')) {
+        const rootURL = this.module.url;
+        const childSpecifier = e.message.match(/'(.*?)'/)[1];
```

Comment:
Since `*` already means "0 or more matches", it seems you can drop the `?` here.

Reply:
Could also go with `/'([^']*)'/` (left to right: match `'`, followed by any character that is not `'` 0 or more times, followed by `'`). If you're always expecting at least one character between the quotes, use `+` instead of `*`.

Reply:
The ? makes the * non-greedy, so it is needed. Also this regex ignores any backslashed quotes. Is that good enough?

How about something like

/‘((?:\\’|[^’])*?)’/

(Apologies for trolling. Could not resist 😀 )

Reply:
I see. I was missing that the full error message is something like

```
The requested module 'foo.mjs' does not provide an export named 'default'
```

(note the second pair of quotes at the end) so non-greediness is indeed needed. TL;DR ignore my first comment :)

Reply:
your first suggestion actually doesn't work 😅. It matches between first and last quote in the entire string (there are a couple). The second suggestion works but I'm unsure the advantage

Reply:
@pipobscure AFAIK we will never get escaped quotes... I'd prefer to keep it small and readable.

Reply:
Is it possible to decorate the error with proper properties for those values? Or would that blow up this PR too much? It feels awkward if we are the ones burying the value in a string and then also are the ones using a regex to get it back.